client/dashboard/sites/actions.tsx

```diff
@@ -52,6 +52,24 @@ export function getActions( router: AnyRouter ): Action< Site >[] {
 			},
 			isEligible: ( item: Site ) => canManageSite( item ),
 		},
+		{
+			id: 'restore',
+			isPrimary: true,
+			icon: backup,
+			label: __( 'Restore site' ),
+			isEligible: ( item: Site ) =>
+				!! item.is_deleted && ! isP2( item ) && ! isSelfHostedJetpackConnected( item ),
```

Comment:
[nit] [note to self] have we considered how to prevent non-admins from seeing the management actions, like this one and Settings action? We can solve that in a follow-up PR.

Reply:
It’s already guarded by `canManageSite`. However, for deleted sites, the `manage_options` capability is not returned as mentioned [here](https://github.com/Automattic/wp-calypso/blob/6bade06edc2a262475f02cd9758aac2c6e5d85b8/client/sites/components/sites-dataviews/actions.tsx#L199-L201).